sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java

```diff
@@ -0,0 +1,238 @@
+import com.azure.cosmos.implementation.changefeed.implementation.ChangeFeedState;
+import com.azure.cosmos.implementation.feedranges.FeedRangeContinuation;
+import com.azure.cosmos.implementation.feedranges.FeedRangeInternal;
+import com.azure.cosmos.implementation.query.CompositeContinuationToken;
+import com.azure.cosmos.util.Beta;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+@Beta(Beta.SinceVersion.WHATEVER_NEW_VERSION)
+public final class CosmosChangeFeedRequestOptions {
+    private static final Integer DEFAULT_MAX_ITEM_COUNT = 1000;
+    private final ChangeFeedState continuationState;
+    private final FeedRangeInternal feedRangeInternal;
+    private final Map<String, Object> properties;
+    private Integer maxItemCount;
+    private ChangeFeedMode mode;
+    private ChangeFeedStartFromInternal startFromInternal;
+
+    private CosmosChangeFeedRequestOptions(
+        FeedRangeInternal feedRange,
+        ChangeFeedStartFromInternal startFromInternal,
+        ChangeFeedMode mode,
+        ChangeFeedState continuationState) {
+
+        super();
+
+        if (feedRange == null) {
+            throw new NullPointerException("feedRange");
+        }
+
+        if (startFromInternal == null) {
+            throw new NullPointerException("startFromInternal");
+        }
+
+        this.maxItemCount = DEFAULT_MAX_ITEM_COUNT;
+        this.feedRangeInternal = feedRange;
+        this.startFromInternal = startFromInternal;
+        this.continuationState = continuationState;
+
+        if (mode != ChangeFeedMode.INCREMENTAL && mode != ChangeFeedMode.FULL_FIDELITY) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Argument 'mode' has unsupported change feed mode %s",
+                    mode.toString()));
+        }
+
+        this.mode = mode;
+        this.properties = new HashMap<>();
+    }
+
+    ChangeFeedState getContinuation() {
+        return this.continuationState;
+    }
+
+    @Beta(Beta.SinceVersion.WHATEVER_NEW_VERSION)
+    public FeedRange getFeedRange() {
+        return this.feedRangeInternal;
+    }
+
+    /**
+     * Gets the maximum number of items to be returned in the enumeration
+     * operation.
+     *
+     * @return the max number of items.
+     */
+    @Beta(Beta.SinceVersion.WHATEVER_NEW_VERSION)
+    public Integer getMaxItemCount() {
+        return this.maxItemCount;
+    }
+
+    /**
+     * Sets the maximum number of items to be returned in the enumeration
+     * operation.
+     *
+     * @param maxItemCount the max number of items.
+     * @return the FeedOptionsBase.
+     */
+    @Beta(Beta.SinceVersion.WHATEVER_NEW_VERSION)
+    public CosmosChangeFeedRequestOptions setMaxItemCount(Integer maxItemCount) {
+        this.maxItemCount = maxItemCount;
+        return this;
+    }
+
+    ChangeFeedMode getMode() {
```

Comment:
Happened independently - will consider it when actually adding the new API on CosmosAsyncContainer and CosmosContainer

Reply:
Which would be the default value then? It is a critical part of consuming the Change Feed and the choice should be on the user. Defaulting to Incremental if the user created the container with FF support will potentially create CRIs because the users would expect the Change Feed to behave with retention and the discoverability of the options makes it so it's rather an optional parameter.

Reply:
Let's take this separately - I kind of disagree because FullFidelity is increasing the risk for data loss. So yes - customers need to make the decision - but I don't see why we can't have a default behavior (like we do for Consistency Model). I don't see FF being used for more than a couple percent of scenarios in the near future. Making the barrier to accidentally use full fidelity (like with allow bulk in .Net - sounds cool so customers just enable it without fully understanding the consequences) high was intentional. Also FullFidelity will never GA before we have to GA CF pull model for Java to support Spark - so I rather keep it separate which allows just marking the withFullFidelity still as @Beta